### PR TITLE
[firebase_remote_config] Support v2 android embedder.

### PR DESCRIPTION
**Pull request by cyanglaz**
## Related Issues

https://github.com/flutter/flutter/issues/41876

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by creativecreatorormaybenot** [comment]
Closes #72.

@cyanglaz I looked through your PR and I cannot find implementations for the retry time.

[From the update article](https://firebase.googleblog.com/2019/08/remote-config-v2-sdk-update.html):

>Another important update for smarter fetching is the introduction of the new **FetchTimeoutInSeconds** and **MinimumFetchIntervalInSeconds** parameters, which can be used together to provide smarter fetching policies for your app. 

Am I missing something?

File: packages/firebase_remote_config/example/android/app/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfigexample/MainActivity.java [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
   @Override
-  protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    GeneratedPluginRegistrant.registerWith(this);
+  public void configureFlutterEngine(FlutterEngine flutterEngine) {
+    super.configureFlutterEngine(flutterEngine);
+    flutterEngine.getPlugins().add(new FirebaseRemoteConfigPlugin());
+
+    ShimPluginRegistry shimPluginRegistry = new ShimPluginRegistry(flutterEngine);
+    FirebaseCorePlugin.registerWith(
```
Comment (xster) [inline packages/firebase_remote_config/example/android/app/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfigexample/MainActivity.java]: ditto from other PR

**Comment by cyanglaz** [comment]
@creativecreatorormaybenot 
This PR is to make the plugin support the Flutter's V2 embedding rather than migrating the plugin to use the firebase V2 SDK.
Does it answer your question?

File: packages/firebase_remote_config/example/android/app/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfigexample/MainActivity.java (hunk 1 above)
Comment (cyanglaz) [reply packages/firebase_remote_config/example/android/app/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfigexample/MainActivity.java]: I might have missed some updates about the V2 embedding. I took a look of other firebase plugin migration PR and figured we don't have to use the `ShimPluginRegistry` to manually register the `firebase core` plugin anymore. And instead we need to add the e2e plugin. So I have updated the file the same. Could you please take another look see if it is good to merge? @xster

Comment (xster) [reply packages/firebase_remote_config/example/android/app/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfigexample/MainActivity.java]: I meant rather that https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_core/android/src/main/java/io/flutter/plugins/firebase/core/FirebaseCorePlugin.java has a v2 embedding implementation. Shouldn't it just work if you did `flutterEngine.getPlugins.add(new FirebaseCorePlugin());`?

Comment (cyanglaz) [reply packages/firebase_remote_config/example/android/app/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfigexample/MainActivity.java]: I see thanks! Will update the dependency as well.

**Pull request merged**